aten/src/ATen/native/cpu/layer_norm_kernel.cpp

```diff
@@ -94,60 +107,159 @@ void LayerNormBackwardKernelImplInternal(
     Tensor* dX,
     Tensor* dgamma,
     Tensor* dbeta) {
+  using Vec = vec256::Vec256<T>;
   DCHECK_EQ(dY.numel(), M * N);
   DCHECK_EQ(X.numel(), M * N);
   DCHECK_EQ(mean.numel(), M);
   DCHECK_EQ(rstd.numel(), M);
   DCHECK(!gamma.defined() || gamma.numel() == N);
-  const T* dY_data = dY.template data_ptr<T>();
-  const T* X_data = X.template data_ptr<T>();
-  const T* mean_data = mean.template data_ptr<T>();
-  const T* rstd_data = rstd.template data_ptr<T>();
-  const T* gamma_data =
-      gamma.defined() ? gamma.template data_ptr<T>() : nullptr;
+  T* dY_data = dY.template data_ptr<T>();
```

Comment:
Not sure why we remove const for all T* here, any reason?

Reply:
ah... took me quite sometime to remember why i did that. `reduce_all` used to accept `scalar_t* input` so passing `const T*` would trigger a compiler error if i called `reduce_all`. Anyway, i updated `reduce_all` signature to be `const scalar_t*`, it makes more sense.